modules/packages/Crypto.chpl

```diff
@@ -0,0 +1,988 @@
+/*
+ * Copyright 2004-2017 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*  A cryptographic library based on OpenSSl, targeted at flexible encryption
+    purposes.
+
+    The Crypto module focuses on providing various cryptographic utilities such as
+
+    - Symmetric Ciphers
+    - Asymmetric Ciphers
+    - Hashing Functions
+    - Key Derivation Functions
+    - Cryptographically-Secure Pseudo-Random Number Generators
+    - Miscellaneous Utilities such as CryptoBuffer, Envelopes, etc.
+
+    .. note::
+
+      This module is currently under development and will expand significantly
+      in upcoming releases.
+
+*/
+
+module Crypto {
+  require "openssl/evp.h", "openssl/pem.h",
+          "openssl/bn.h", "openssl/bio.h",
+          "openssl/aes.h", "openssl/rand.h", "-lcrypto";
+  //require "CryptoHandlers/rsa_complex_bypass_handler.h";
+
+  pragma "no doc"
```

Comment:
When interfacing with C libraries, there is a common pattern to consolidate the C interface into a submodule, typically named `C_<library>`, e.g. [BLAS.C_BLAS](http://chapel.cray.com/docs/master/modules/packages/BLAS/C_BLAS.html) of the [BLAS](http://chapel.cray.com/docs/master/modules/packages/BLAS.html) module.

This might be a nice way to keep these `extern` symbols documented, but also separated from the intended user-facing documentation, e.g. `Crypto.C_OpenSSL`.

Reply:
Where does this file go? And how can I import this into our main module?

Reply:
It can be a sub-module of the Crypto module.  In the BLAS/C_BLAS example @ben-albrecht linked, `module C_BLAS` is defined near the end of the file as a sub-module of the main `BLAS` module.  `BLAS` then does `use C_BLAS;` at the beginning of the module.

Reply:
Okay. Cool. I'll get on to it.